pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -260,6 +279,28 @@ func (c *Reconciler) reconcile(ctx context.Context, r *v1alpha1.Route) error {
 	return nil
 }
 
+func (c *Reconciler) reconcileDeletion(ctx context.Context, r *v1alpha1.Route) error {
+	logger := logging.FromContext(ctx)
+
+	// If our Finalizer is first, delete the ClusterIngress for this Route
+	// and remove the finalizer.
+	if len(r.Finalizers) == 0 || r.Finalizers[0] != routeFinalizer {
+		return nil
+	}
+
+	// Delete the ClusterIngress resources for this Route.
+	logger.Info("Cleaning up ClusterIngress")
+	if err := c.deleteClusterIngressesForRoute(r); err != nil {
```

Comment:
If this succeeds for some reason and the ingress goes away, will (see previous comment) subsequent calls return an error? If so, if anything goes wrong between success here and removing the route finalizer, seems like a route could get stuck in a state where it actually never gets deleted? I guess, point being, can this return a NotFound and if so, should it be handled here instead of just returning a notfound error and getting a route stuck?

Reply:
see above.

Reply:
Awesome, thanks so much for checking!